data/json/furniture_and_terrain/terrain-bridges-docks.json

```diff
@@ -216,5 +216,108 @@
       "ter_set": "t_water_dp",
       "items": [ { "item": "rock", "count": [ 4, 20 ] }, { "item": "rebar", "count": [ 10, 20 ] } ]
     }
+  },
+  {
+    "type": "terrain",
+    "id": "t_metal_dock",
+    "name": "dock",
+    "description": "A metal platform held by a support made of concrete set into the ground.",
+    "symbol": "8",
+    "looks_like": "t_metal_floor",
+    "color": "gray",
+    "move_cost": 2,
+    "flags": [ "TRANSPARENT", "FLAT" ],
+    "deconstruct": { "ter_set": "t_water_sh", "items": [ { "item": "nail", "charges": [ 6, 12 ] }, { "item": "2x4", "count": 8 } ] },
+    "bash": {
+      "str_min": 20,
+      "str_max": 80,
+      "sound": "smash!",
+      "sound_fail": "whump!",
+      "ter_set": "t_water_moving_sh",
+      "items": [ { "item": "sheet_metal", "count": 2, "prob": 35 }, { "item": "scrap", "count": [ 2, 4 ] } ]
+    }
+  },
+  {
+    "type": "terrain",
+    "id": "t_concrete_dock_deep_pile",
+    "name": "deep dock pile",
+    "description": "A sturdy pile made of poured concrete extending all the way to the riverbed.  A metal frame will be required to support a surface.",
+    "looks_like": "t_concrete_wall",
+    "symbol": "|",
+    "color": "gray",
+    "move_cost": 8,
+    "flags": [ "TRANSPARENT", "LIQUID", "SWIMMABLE", "DEEP_WATER", "FISHABLE" ],
+    "deconstruct": { "ter_set": "t_water_moving_dp", "items": [ { "item": "spike", "count": 4 }, { "item": "rock", "count": [ 6, 10 ] } ] },
```

Comment:
Would rebar replace spikes or be in addition to spikes? Rebar might also not be used in concrete piles.

Reply:
updated